glusterd2/commands/volumes/volume-options-get.go

```diff
@@ -0,0 +1,99 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/glusterd2/xlator/options"
+	"github.com/gluster/glusterd2/pkg/api"
+
+	"github.com/gorilla/mux"
+)
+
+func volumeOptionsGetHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+
+	volname := mux.Vars(r)["volname"]
+
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+
+	optname, found := mux.Vars(r)["optname"]
+
+	if found {
+		opt, err := xlator.FindOption(optname)
+		if err != nil {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, err)
+			return
+		}
+		resp := createVolumeOptionGetResp(volinfo, opt)
+		restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+	} else {
+		resp := createVolumeOptionsGetResp(volinfo)
+		restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+	}
+}
+
+func createVolumeOptionGetResp(volinfo *volume.Volinfo, opt *options.Option) *api.VolumeOptionGetResp {
+	var resp api.VolumeOptionGetResp
+
+	for _, xl := range xlator.Xlators() {
+		modified, found := false, false
+		val := opt.DefaultValue
+
+		for _, k := range opt.Key {
+			if val, found = volinfo.Options[xl.ID+"."+k]; found {
+				modified = found
+			}
+			resp = api.VolumeOptionGetResp{
+				OptName:      xl.ID + "." + k,
+				Value:        val,
+				Modified:     modified,
+				DefaultValue: opt.DefaultValue,
+				OptLevel:     api.OptionLevel(opt.Level).String(),
+			}
+			break
+		}
+	}
+
+	return &resp
+}
+
+func createVolumeOptionsGetResp(volinfo *volume.Volinfo) *api.VolumeOptionsGetResp {
+	var resp api.VolumeOptionsGetResp
+
+	for _, xl := range xlator.Xlators() {
+		// TODO Once we have information on supported xlators
+		// per volume type we can filter out these options. For
+		// now return all options
+
+		for _, opt := range xl.Options {
+			var val string
+
+			for _, k := range opt.Key {
+				val = opt.DefaultValue
+				modified := false
+
+				if _, found := volinfo.Options[k]; found {
+					val = volinfo.Options[k]
+					modified = true
+				}
```

Comment:
I see an opportunity for code reuse here. `createVolumeOptionsGetResp` can may be call `createVolumeOptionGetResp` in a loop. Let me know what you think.

Reply:
initially, I just did resolved conflicts of  #572.  now i think there is no code duplication